libs/@hashintel/type-editor/src/shared/data-types-options-context.tsx

```diff
@@ -199,28 +204,47 @@ const getArrayDataTypeDisplay = (
     if (!dataType.prefixItems) {
       return expectedValuesDisplayMap.emptyList;
     }
+
     items = dataType.prefixItems;
+
+    const itemTypes = new Set<SimpleTypedValueSchema["type"]>();
+
+    for (const item of items) {
+      if ("anyOf" in item) {
+        /**
+         * @todo H-3373 support tuple data types which can have multiple different values in a position
+         */
+        return expectedValuesDisplayMap.mixedArray;
+      }
+
+      itemTypes.add(item.type);
+    }
+
+    if (itemTypes.size === 1) {
+      /**
+       * @todo H-3373 support tuple data types with a fixed number of single values
+       */
+      return expectedValuesDisplayMap[
+        `${itemTypes.values().next().value}Array` as keyof typeof expectedValuesDisplayMap
+      ];
+    }
+
+    /**
+     * @todo H-3373 support tuple data types with different types of values in each position
+     */
+    return expectedValuesDisplayMap.mixedArray;
+
     // TODO: Remove when `const` is not allowed on arrays
     //   see https://linear.app/hash/issue/H-3368/remove-const-from-array-constraints
   } else if (!isArrayConstraints(dataType)) {
     return expectedValuesDisplayMap.emptyList;
   } else if (!dataType.items) {
     return expectedValuesDisplayMap.mixedArray;
+  } else if ("anyOf" in dataType.items) {
+    return expectedValuesDisplayMap.mixedArray;
```

Comment:
@CiaranMn Do we want to use the same approach for `anyOf` as for `prefixItems`? because for example this is still a number array:
```json5
{
	anyOf: [
		{
			type: "number",
			multipleOf: 2
		},
		{
			type: "number",
			multipleOf: 3
		},
	]
}
```

Reply:
Yes, we can change this, it will help as a marker for future work

Reply:
Alright, I'll apply the changes

Reply:
Thank you!

Reply:
Applied the changes in [`eda3a5a` (#5232)](https://github.com/hashintel/hash/pull/5232/commits/eda3a5ae8069c97b55ecb47395921c53718da46f)

Reply:
Well, `anyOf` is not allowed in `items` anymore anyway 😅